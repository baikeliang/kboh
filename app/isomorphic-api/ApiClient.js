import 'babel-polyfill'
import Promise from 'bluebird'
import fetch from 'isomorphic-fetch'
import 'url-search-params-polyfill'

console.log('ssaVVVVVVVVVVV!!!!!!!')
if(typeof window !=="undefined")
console.log(window.URLSearchParams)

var URLSearchParams = (typeof window !=="undefined")?window.URLSearchParams:require('urlsearchparams').URLSearchParams;
//import { URLSearchParams } from 'urlsearchparams'

const methods = ['GET', 'POST'];
export default class ApiClient {
    constructor(args) {
        // code
        methods.forEach((method) =>
            this[method] = (path, { params, data } = {} ,{format,done,error}) => {
                ///////////////////////////////
                var u = new URLSearchParams();
                console.log(method)
                var req = {
                    method: method,
                    mode: 'cors',
                    cache: 'default'
                }
                if ((method == 'GET') && params) {
                    for (var key of Object.keys(params)) {
                      u.append(key, params[key])
                    }
                } else if (method == 'POST' && data) {

                    req.body = JSON.stringify(data)
                    req.headers = {
                        'Content-Type': 'application/json'
                    }

                }
                console.log('8888888888'+path+u)
                return fetch(path + u, req).then(format).then(done,error)
            }
        )
    }

    // methods
}
