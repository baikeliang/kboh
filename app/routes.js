import React from 'react';


import { Router, DefaultRoute, Route, NotFoundRoute, Redirect } from 'react-router';

import ToOrder from 'app/toorder/main.js'

import UserCenter from 'app/usercenter/main.js'

import MyOrders from 'app/usercenter/myorder/myorders.js'
import MyOrderInfo from 'app/usercenter/myorder/myorderinfo.js'
import MyCases from 'app/usercenter/mycase/mycases.js'
import MyCaseDesc from 'app/usercenter/mycase/mycasedesc.js'
import Login from 'app/usercenter/login.js'


const App = ({ children }) => (
  <div>
    {children || "ddddd"} 
  </div>
)

function requireAuth(nextState, replaceState) {
  /*
  if ((__SERVER__==false)&&localStorage&&(!localStorage.token))
      replaceState({ nextPathname: nextState.location.pathname }, '/login')
  */
  
  console.log("require.....auth")
  
}

const routes = (
      <Route path="/" component={ App }>
        <Route path="usercenter(?)" component={ UserCenter } onEnter={requireAuth}>
          <Route path="toOrder"   component={ ToOrder }/>
          <Route path="myOrders"  component={ MyOrders } />
          <Route path="myOrderInfo" component={ MyOrderInfo } />
          <Route path="myCases"   component={ MyCases }/>
          <Route path="myCaseDesc"    component={ MyCaseDesc }/>
        </Route>        
        <Route path="login" component={ Login } />  
      </Route>
)

export default routes

