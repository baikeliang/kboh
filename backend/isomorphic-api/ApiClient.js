import 'babel-polyfill'
import Promise from 'bluebird'
//import fetch from 'isomorphic-fetch'
import 'isomorphic-fetch'
import 'url-search-params-polyfill'

var URLSearchParams = (typeof window !=="undefined")?window.URLSearchParams:require('urlsearchparams').URLSearchParams;
//import { URLSearchParams } from 'urlsearchparams'

const methods = ['GET', 'POST'];
export default class ApiClient {
    constructor(args) {
        // code
        methods.forEach((method) =>
            this[method] = (path, { params, data } = {} ,{format,done,error}) => {
                ///////////////////////////////
                var u = new URLSearchParams();
                var req = {
                    method: method,
                    mode: 'cors',
                    cache: 'default'
                }
                if ((method == 'GET') && params) {
                    for (var key of Object.keys(params)) {
                      u.append(key, params[key])
                    }
                } else if (method == 'POST' && data) {
                    console.log(data)
                    console.log("pppppppp")
                    req.body = JSON.stringify(data)
                    console.log(req.body)
                    req.headers = {

                    };

                }
                return fetch(path + u, req).then(format).then(done,error)
            }
        )
    }

    // methods
}
