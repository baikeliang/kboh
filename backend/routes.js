import React from 'react';


import { Router, DefaultRoute, Route, NotFoundRoute, Redirect,IndexRoute} from 'react-router';


import DashBoard  from 'backend/useradmin/main.js'

import Login from 'backend/login.js'


const App = ({ children }) => (
  <div>
    {children || "ddddd"}
  </div>
)

const UserAdmin =  ({ children }) => (
  <div>
    {children}
  </div>
)

function requireAuth(nextState, replaceState) {

  //if ((__SERVER__==false)&&localStorage&&(!localStorage.token))
  //    replaceState({ nextPathname: nextState.location.pathname }, '/login')


  console.log("require.....auth")

}

const routes = (
      <Route path="/" component={ App }>
        <Route path="useradmin(?)" component={ UserAdmin } onEnter={requireAuth}>
          <IndexRoute component={ DashBoard }/>
        </Route>
        <Route path="login" component={ Login } />
      </Route>
)

export default routes

