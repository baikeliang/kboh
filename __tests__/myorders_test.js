import Promise from 'bluebird';
import React from 'react';
import {
  Provider,
  connect
} from 'react-redux';
import {
  Router,
  createMemoryHistory,
  match,
  Route,
  IndexRoute
} from 'react-router';
import {
  createStore,
  combineReducers
} from 'redux';
import { combineReducers as combineImmutableReducers } from 'redux-immutable';
import {
  mount,
  render
} from 'enzyme';
import { spy } from 'sinon';
import { default as Immutable } from 'immutable';

import routes from 'app/routes.js'

import {
  configureStore,
  DevTools
} from 'app/configure-store'

import ApiClient from 'app/isomorphic-api/ApiClient'

import {
  orders_is_not_empty,
  orders_is_empty,
  orders_is_not_valid,
  orders_response_error_500,
  resetMock as resetMockOrder
} from '../__mocks__/server_getorders_mocker.js'

import {
  auth_success,
  auth_is_not_valid,
  auth_response_error_500,
  auth_success_serverrender_openid,
  resetMock as resetMockAuth
} from '../__mocks__/server_auth_mocker.js'

import {
  endGlobalLoad,
  beginGlobalLoad
} from '../node_modules/redux-connect/lib/store';
import AsyncConnect from '../node_modules/redux-connect/lib/components/AsyncConnect';

import {
  mountToJson,
  renderToJson
} from 'enzyme-to-json';

import { loadOnServer } from 'redux-connect'


describe('top', function suite() {
  const endGlobalLoadSpy = spy(endGlobalLoad);
  const beginGlobalLoadSpy = spy(beginGlobalLoad);

  const ReduxAsyncConnect = connect(null, {
    beginGlobalLoad: beginGlobalLoadSpy,
    endGlobalLoad: endGlobalLoadSpy,
  })(AsyncConnect);


  pit("auth not valid",function test(){

    window.__SERVER__ = false;

    var preloadstate = Immutable.fromJS({})

    var client = new ApiClient()

    const history = createMemoryHistory();

    var store = configureStore(history, client, preloadstate)

    const proto = ReduxAsyncConnect.WrappedComponent.prototype;

    spy(proto, 'loadAsyncData');
    spy(proto, 'componentDidMount');
    var reloadOnPropsChange = (props, nextProps) => {
        // reload only when path/route has changed
        return props.location.pathname !== nextProps.location.pathname;
    };
    var location = history.createLocation('/usercenter');

    history.push('/usercenter/myOrders');

    auth_is_not_valid();

    var wrapper = mount(
                  <Provider store={store} key="provider">
                      <Router routes={routes} location={location} render={(props) => {
                          return (<ReduxAsyncConnect {...props} helpers={{client}} filter={item => !item.deferred} reloadOnPropsChange={reloadOnPropsChange}/>)}
                      }  history={history}>
                      </Router>
                  </Provider>
                 );


    expect(proto.loadAsyncData.calledOnce).toBe(true);
    expect(proto.componentDidMount.calledOnce).toBe(true);



    expect(beginGlobalLoadSpy.called).toBe(true);
    beginGlobalLoadSpy.reset();

    return proto.loadAsyncData.returnValues[0].then(() => {
      expect(endGlobalLoadSpy.called).toBe(true);
      var state = store.getState().toJS();
      expect(!!state.auth.user).toBe(false)
      expect(mountToJson(wrapper.find('Login div'))).toMatchSnapshot();
      endGlobalLoadSpy.reset();
      proto.loadAsyncData.restore();
      proto.componentDidMount.restore();
      resetMockOrder();
      resetMockAuth();
    });




  })

  pit("auth response 500",function test(){

    window.__SERVER__ = false;

    var preloadstate = Immutable.fromJS({})

    var client = new ApiClient()

    const history = createMemoryHistory();

    var store = configureStore(history, client, preloadstate)

    const proto = ReduxAsyncConnect.WrappedComponent.prototype;

    spy(proto, 'loadAsyncData');
    spy(proto, 'componentDidMount');
    var reloadOnPropsChange = (props, nextProps) => {
        // reload only when path/route has changed
        return props.location.pathname !== nextProps.location.pathname;
    };
    var location = history.createLocation('/usercenter');

    history.push('/usercenter/myOrders');

    auth_response_error_500();

    var wrapper = mount(
                  <Provider store={store} key="provider">
                      <Router routes={routes} location={location} render={(props) => {
                          return (<ReduxAsyncConnect {...props} helpers={{client}} filter={item => !item.deferred} reloadOnPropsChange={reloadOnPropsChange}/>)}
                      }  history={history}>
                      </Router>
                  </Provider>
                 );


    expect(proto.loadAsyncData.calledOnce).toBe(true);
    expect(proto.componentDidMount.calledOnce).toBe(true);



    expect(beginGlobalLoadSpy.called).toBe(true);
    beginGlobalLoadSpy.reset();

    return proto.loadAsyncData.returnValues[0].then(() => {
      expect(endGlobalLoadSpy.called).toBe(true);
      var state = store.getState().toJS();
      expect(!!state.auth.user).toBe(false)
      //we should have a 500 page
      endGlobalLoadSpy.reset();
      proto.loadAsyncData.restore();
      proto.componentDidMount.restore();
      resetMockOrder();
      resetMockAuth();
    });




  })



  pit("auth valid : not loading: orders not empty ",function test(){

    window.__SERVER__ = false;

    var preloadstate = Immutable.fromJS({})

    var client = new ApiClient()

    const history = createMemoryHistory();

    var store = configureStore(history, client, preloadstate)

    const proto = ReduxAsyncConnect.WrappedComponent.prototype;

    spy(proto, 'loadAsyncData');
    spy(proto, 'componentDidMount');
    var reloadOnPropsChange = (props, nextProps) => {
        // reload only when path/route has changed
        return props.location.pathname !== nextProps.location.pathname;
    };
    var location = history.createLocation('/usercenter');

    history.push('/usercenter/myOrders');

    auth_success();
    orders_is_not_empty();

    var wrapper = mount(
                  <Provider store={store} key="provider">
                      <Router routes={routes} location={location} render={(props) => {
                          return (<ReduxAsyncConnect {...props} helpers={{client}} filter={item => !item.deferred} reloadOnPropsChange={reloadOnPropsChange}/>)}
                      }  history={history}>
                      </Router>
                  </Provider>
                 );


    expect(proto.loadAsyncData.calledOnce).toBe(true);
    expect(proto.componentDidMount.calledOnce).toBe(true);



    expect(beginGlobalLoadSpy.called).toBe(true);
    beginGlobalLoadSpy.reset();

    return proto.loadAsyncData.returnValues[0].then(() => {
      expect(endGlobalLoadSpy.called).toBe(true);
      var state = store.getState().toJS();
      expect(!!state.auth.user).toBe(true)
      console.log(wrapper)
      expect(mountToJson(wrapper.find('MyOrders div'))).toMatchSnapshot();
      endGlobalLoadSpy.reset();
      proto.loadAsyncData.restore();
      proto.componentDidMount.restore();
      resetMockOrder();
    });


  })

  pit("server render auth succed with openid",function test(){

      window.__SERVER__ = true;

      var preloadstate = Immutable.fromJS({})

      var client = new ApiClient()

      const history = createMemoryHistory();

      var store = configureStore(history, client, preloadstate)

      const proto = ReduxAsyncConnect.WrappedComponent.prototype;

      spy(proto, 'loadAsyncData');

      auth_success_serverrender_openid();

    return new Promise((resolve, reject) => {

      match({ history, routes, location: '/usercenter/myOrders' }, (error, redirectLocation, renderProps) => {

         if (error) {
           return reject(error);
         }
         if (redirectLocation) {
           return reject(new Error('redirected'));
         }
         if (!renderProps) {
           return reject(new Error('404'));
         }

         var req = {
            openid:'openidfortest',
            cookies:{
              tokenbohe:'dummy'
            }
         }

        return loadOnServer({ ...renderProps, store, params:{ req }}).then(() => {
            const content = render(
              <Provider store={store} key="provider">
                <ReduxAsyncConnect {...renderProps} />
              </Provider>
            )
            var state = store.getState().toJS();
            expect(!!state.auth.user).toBe(true);
            expect(renderToJson(content)).toMatchSnapshot();
            resolve();
          },(err)=>{

       }).catch(reject);

    })
   })
  })

})

