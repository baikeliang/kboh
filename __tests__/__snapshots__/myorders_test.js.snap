exports[`top auth not valid 1`] = `
<div
  style={
    Object {
      "FontFamily": "微软雅黑,Arial,sans-serif",
      "WebkitTapHighlightColor": "transparent",
      "background": "#fff",
      "color": "#000",
      "fontFamily": "Helvetica Neue, Helvetica, Arial, sans-serif",
      "fontSize": "14px",
      "height": "100%",
      "margin": 0,
      "padding": 0,
      "position": "relative",
    }
  }>
  <div
    className={undefined}>
    <img
      alt=""
      src=""
      style={
        Object {
          "display": "block",
          "margin": "0 auto",
          "maxWidth": "200px",
          "paddingBottom": "44px",
          "paddingTop": "39px",
          "width": "40%",
        }
      } />
  </div>
  <Hammer
    onSwipe={[Function]}>
    <div
      className={undefined}>
      <div
        className={undefined}>
        <a
          className={
            Object {
              "color": "#1b7c86",
            }
          }
          href="#">
          验证码登录
        </a>
        <a
          href="#">
          密码登录
        </a>
        <div
          className={undefined} />
      </div>
      <ReactSwipe
        className=""
        style={
          Object {
            "child": Object {
              "float": "left",
              "position": "relative",
              "transitionProperty": "transform",
              "width": "100%",
            },
            "container": Object {
              "overflow": "hidden",
              "position": "relative",
              "visibility": "hidden",
            },
            "wrapper": Object {
              "overflow": "hidden",
              "position": "relative",
            },
          }
        }
        swipeOptions={
          Object {
            "auto": 0,
            "callback": [Function],
            "continuous": false,
            "disableScroll": false,
            "speed": 300,
            "startSlide": 0,
            "transitionEnd": [Function],
          }
        }>
        <div
          className="react-swipe-container "
          id={undefined}
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
              "visibility": "hidden",
            }
          }>
          <div
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }>
            <div
              style={
                Object {
                  "float": "left",
                  "position": "relative",
                  "transitionProperty": "transform",
                  "width": "100%",
                }
              }>
              <div
                className={undefined}>
                <div
                  className="undefined undefined">
                  <input
                    className="undefined tel phone"
                    id="phone"
                    maxLength="11"
                    placeholder="手机号"
                    type="tel" />
                  <div
                    className={undefined} />
                </div>
                <div
                  className="undefined undefined">
                  <input
                    className={undefined}
                    id="code"
                    maxLength="6"
                    placeholder="验证码"
                    type="text" />
                  <input
                    className={undefined}
                    id="sendYzm"
                    type="button"
                    value="发送验证码" />
                </div>
                <div
                  className={undefined}>
                  <input
                    className={undefined}
                    id="loginBut"
                    type="button"
                    value="登录" />
                  <p
                    className={undefined}>
                    <a
                      href="javascript:;"
                      id="go_register">
                      立即注册
                    </a>
                  </p>
                </div>
              </div>
            </div>
            <div
              style={
                Object {
                  "float": "left",
                  "position": "relative",
                  "transitionProperty": "transform",
                  "width": "100%",
                }
              }>
              <div
                className={undefined}>
                <div
                  className="undefined undefined">
                  <input
                    className={undefined}
                    id="phone2"
                    maxLength="11"
                    placeholder="手机号"
                    type="tel" />
                  <div
                    className={undefined} />
                </div>
                <div
                  className="undefined undefined">
                  <input
                    className={undefined}
                    id="pwd"
                    maxLength="20"
                    placeholder="6～20位密码"
                    type="password" />
                  <a
                    className={undefined}
                    href="javascript:;" />
                  <a
                    className={undefined}
                    href="javascript:;" />
                  <i
                    className="forget">
                    忘记密码?
                  </i>
                </div>
                <div
                  className="undefined undefined">
                  <input
                    className={undefined}
                    id="loginBut2"
                    type="button"
                    value="登录" />
                  <p
                    className={undefined}>
                    <a
                      href="javascript:;"
                      id="go_register">
                      立即注册
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </ReactSwipe>
    </div>
  </Hammer>
</div>
`;

exports[`top auth response 500 1`] = `""`;

exports[`top auth valid : not loading: orders not empty  1`] = `
<div>
  <div
    className={undefined}
    style={
      Object {
        "display": "none",
      }
    }>
    <dl>
      <dt>
        <img
          alt=""
          src="" />
      </dt>
      <dd>
        您目前还没有进行任何预约
      </dd>
    </dl>
    <div
      className={undefined}>
      <img
        alt=""
        src="" />
      <p>
        轻快预约   从“齿”简单
      </p>
    </div>
  </div>
  <Hammer
    direction="DIRECTION_VERTICAL"
    onPan={[Function]}
    options={
      Object {
        "touchAction": "pan-y",
      }
    }>
    <div>
      <div
        className={undefined}
        style={
          Object {
            "marginTop": "10px",
          }
        }>
        <ReactList
          axis="y"
          itemRenderer={[Function]}
          itemsRenderer={[Function]}
          length={11}
          pageSize={10}
          threshold={100}
          type="uniform"
          useStaticSize={false}
          useTranslate3d={false}>
          <div
            style={
              Object {
                "position": "relative",
              }
            }>
            <div
              style={
                Object {
                  "WebkitTransform": "translate(0px, 0px)",
                  "msTransform": "translate(0px, 0px)",
                  "transform": "translate(0px, 0px)",
                }
              }>
              <div>
                <Connect(MyOrder)
                  id={90}
                  idx={0}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={0}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={1}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={1}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={2}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={2}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={3}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={3}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={4}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={4}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={5}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={5}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={6}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={6}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={7}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={7}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={8}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={8}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
                <Connect(MyOrder)
                  id={90}
                  idx={9}
                  is_return={0}
                  patient_name="tangxin"
                  reserve_number="cool"
                  service_name="baya"
                  status={88}
                  visit_time="23212">
                  <MyOrder
                    id={90}
                    idx={9}
                    is_return={0}
                    load={[Function]}
                    patient_name="tangxin"
                    pushState={[Function]}
                    reserve_number="cool"
                    service_name="baya"
                    status={88}
                    visit_time="23212">
                    <div
                      onClick={[Function]}>
                      <div
                        className="undefined undefined undefined">
                        <a
                          className={undefined}>
                          <div
                            className={undefined}>
                            <p>
                              <b>
                                <em />
                              </b>
                              <label />
                              <label>
                                <span>
                                  23212
                                </span>
                              </label>
                            </p>
                          </div>
                          <span
                            className={undefined} />
                        </a>
                      </div>
                    </div>
                  </MyOrder>
                </Connect(MyOrder)>
              </div>
            </div>
          </div>
        </ReactList>
      </div>
    </div>
  </Hammer>
</div>
`;
